token/cli/src/main.rs

```diff
@@ -1891,6 +1911,164 @@ async fn command_sync_native(
+            )?,
+        );
+    }
+    let tx_return = handle_tx(
+        &CliSignerInfo {
+            signers: bulk_signers,
+        },
+        config,
+        false,
+        0,
+        instructions,
+    )
+    .await?;
+    Ok(match tx_return {
+        TransactionReturnData::CliSignature(signature) => {
+            config.output_format.formatted_string(&signature)
+        }
+        TransactionReturnData::CliSignOnlyData(sign_only_data) => {
+            config.output_format.formatted_string(&sign_only_data)
+        }
+    })
+}
+
+// Checks if reallocation needed for new specified extensions. If needed returns reallocate instruction
+async fn reallocate_if_needed(
+    program_id: &Pubkey,
+    token_account_pubkey: &Pubkey,
+    owner_pubkey: &Pubkey,
+    new_extension_types: Vec<ExtensionType>,
+    config: &Config<'_>,
+) -> Result<Option<Instruction>, Error> {
+    let token_account = config
+        .rpc_client
+        .get_token_account(token_account_pubkey)
+        .await?
+        .expect("Invalid token account pubkey");
+    let mut existing_extensions: Vec<ExtensionType> = token_account
+        .extensions
+        .iter()
+        .map(parse_ui_extension)
+        .collect();
+    existing_extensions.extend_from_slice(&new_extension_types);
+    let needed_account_len = ExtensionType::get_account_len::<Account>(&existing_extensions);
+    let current_account_len = config
+        .rpc_client
+        .get_account(token_account_pubkey)
+        .await?
+        .data
+        .len();
+    // If reallocation is not needed
+    if needed_account_len <= current_account_len {
+        return Ok(None);
+    }
+    let instruction = reallocate(
+        program_id,
+        token_account_pubkey,
+        &config.fee_payer,
+        owner_pubkey,
+        &config.multisigner_pubkeys,
+        &existing_extensions,
+    )?;
+    Ok(Some(instruction))
+}
+
+fn parse_ui_extension(ui_extension: &UiExtension) -> ExtensionType {
+    match &ui_extension {
+        UiExtension::Uninitialized => ExtensionType::Uninitialized,
+        UiExtension::TransferFeeConfig(_) => ExtensionType::TransferFeeConfig,
+        UiExtension::TransferFeeAmount(_) => ExtensionType::TransferFeeAmount,
+        UiExtension::MintCloseAuthority(_) => ExtensionType::MintCloseAuthority,
+        UiExtension::ConfidentialTransferMint(_) => ExtensionType::ConfidentialTransferMint,
+        UiExtension::ConfidentialTransferAccount(_) => ExtensionType::ConfidentialTransferAccount,
+        UiExtension::DefaultAccountState(_) => ExtensionType::DefaultAccountState,
+        UiExtension::ImmutableOwner => ExtensionType::ImmutableOwner,
+        UiExtension::MemoTransfer(_) => ExtensionType::MemoTransfer,
+        UiExtension::NonTransferable => ExtensionType::NonTransferable,
+        UiExtension::InterestBearingConfig(_) => ExtensionType::InterestBearingConfig,
+        UiExtension::UnparseableExtension => panic!("Unparsable extension"),
+    }
+}
```

Comment:
we can delete this when `get_extension_types` replaces the current parse method

Reply:
resolved with d37c851